File: sdks/php/src/Coupons/Requests/ListCouponsRequest.php
<?php

namespace NovaBilling\Coupons\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class ListCouponsRequest extends JsonSerializableType
{
    /**
     * @var ?bool $isActive
     */
    public ?bool $isActive;

    /**
     * @var ?float $page
     */
    public ?float $page;

    /**
     * @var ?float $limit
     */
    public ?float $limit;

    /**
     * @param array{
     *   isActive?: ?bool,
     *   page?: ?float,
     *   limit?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->isActive = $values['isActive'] ?? null;
        $this->page = $values['page'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/Coupons/Requests/ListCouponsRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListCouponsRequest
{
    [JsonIgnore]
    public bool? IsActive { get; set; }

    [JsonIgnore]
    public double? Page { get; set; }

    [JsonIgnore]
    public double? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
